src/components/App.js

```diff
@@ -9,14 +9,36 @@ import MyLikes from "./MyLikes";
 import AnyUsersPosts from "./AnyUsersPosts";
 import HashtagPosts from './HashtagPosts';
 import MyPosts from "./MyPosts"
-
-
+import isValidUserState from '../helperFunctions/isValidUserState';
 
 export default function App() {
 
-    const [user, setUser] = useState("");
+    const [user, setUser] = useState(null);
 
-    return (
+    const UserStorage = localStorage.getItem("user");
+    const history = useHistory();
+    const path = useLocation().pathname;
+
+    useEffect(() => {
+        if (UserStorage !== null) {
+            const localUserState = JSON.parse(UserStorage);
+            if (isValidUserState(localUserState)){
+                setUser(localUserState);
+                if (path === "/sign-up") history.push("/timeline")
+                else history.push(path);
+            }
+            else {
+                localStorage.clear();
+                setUser(undefined);
+                history.push("/");
+            }
+        } else {
+            setUser(undefined);
```

Comment:
Aqui é possível setar para null?

Reply:
Com a logica atual n. O return do componente retorna uma string vazia se user for null mas renderiza a pagina se user for qualquer outra coisa. Entao eu estou dando setUser(undefined) pra dizer: "olha, n tem nenhum login persistente la no local storage. e se tiver, ele ta corrompido pq falhou a validação então eu apaguei ele. Pode seguir pra tela de login"

Reply:
Se eu fizer setUser(null) no lugar de setUser(undefined) o que vai acontecer eh que o componente App sempre vai renderizar uma string vazia